modAionImpl/src/org/aion/zero/impl/blockchain/AionPendingStateImpl.java

```diff
@@ -434,29 +435,64 @@ public synchronized AionBlock getBestBlock() {
      * libAion uses to work with timers
      */
     @Override
-    public synchronized List<AionTransaction> addPendingTransaction(AionTransaction tx) {
-
-        return addPendingTransactions(Collections.singletonList(tx));
+    public synchronized TxResponse addPendingTransaction(AionTransaction tx) {
+        return addPendingTransactions(Collections.singletonList(tx)).get(0);
```

Comment:
It should not be the case that a transaction gets multiple responses. That would be a mistake.

The idea is that if addPendingTransactions() is called with a list of _n_ transactions, then the return object should have _n_ TxResponses, corresponding one-to-one with the inputs. If we pull any transactions from the cache during execution, those txs do not have their responses returned (which makes sense to me, because they're not relevant as a "return").